src/server/db_slice.h

```diff
@@ -150,6 +186,15 @@ class DbSlice {
   OpResult<PrimeIterator> Find(const Context& cntx, std::string_view key,
                                unsigned req_obj_type) const;
 
+  struct ItAndUpdater {
+    PrimeIterator it;
+    AutoPostUpdate post_updater;
+  };
```

Comment:
Maybe I missed something, but can't we in the future just add it to the PrimeIterator itself? Or are there cases where this is undesirable? I mean the iterator can even track if it was mutably dereferenced

Reply:
That's a very interesting proposal, that I thought about quite a lot. While I would have loved to do it, I think that it might be problematic because we copy `PrimeIterator` all over the place, and that might create "nested" calls such as `PreUpdate()`, `PreUpdate()`, `PostUpdate()`, `PostUpdate()`.
We could maybe handle them (although `PreUpdate()` calls external callbacks, so it's harder to keep in mind), but perhaps the explicit nature of this PR has some advantage to it..